### DISCUSSION TITLE
Django Tenant Context Missing

### DISCUSSION THREAD

**Discussion by matiasgatti** [eternnoir/pyTelegramBotAPI#1984] [Q&A] [answered]
Hi ! I like this app very much. I was able to set Webhook, Receive Message, Send Message y a Multibot enviroment. Everything is good. Only missing one thing: I lost tenant context(postgre schema) when I am on any "command". 

Here is my code:

```
class TelegramWebhook(View):
	def post(self, request, *args, **kwargs):
		received_json_data = json.loads(request.body)
		update = Update.de_json(received_json_data)
		bot = TeleBot(request.tenant.telegram_token)
		register_handlers(bot)
		bot.process_new_messages([update.message])
		return JsonResponse({"ok": "POST request processed"})

def register_handlers(bot: TeleBot, cliente, schema_name):
	bot.register_message_handler(command_start, commands=['start', 'menu'], pass_bot=True)
	bot.register_message_handler(command_novedades, commands=['novedades'], pass_bot=True)

def command_start(message: types.Message, bot: TeleBot):
	### CONTEXT MISSING HERE ###
	clients = Cliente.objects.all() # > Error. Missing Context to access DB
    bot.send_message(message.chat.id, 'This is the Start Menu')
```

IS THERE ANY WAY TO KEEP MY DJANGO-TENANTS CONTEXT BETWEEN FUNCTIONS?  Thanks.

**Comment by matiasgatti** [comment, marked as answer]
Answer my self:

I was able to save current schema name in message like this:

update.message.current_schema_name = current_schema_name
bot.process_new_messages([update.message])

Then I can get the schema name in def command and use it with: with schema_context


(Now I am having issues with register_next_step on this multibot enviroment)

**Comment by coder2020official** [comment]
Sup, sorry for a delayed answer. Instead of assigning schema to Update type, you could use class-based middlewares and pass the schema to the handler

**Reply by matiasgatti** [reply]
Ok. I will try this. I guess this is the best practice approach. Thanks!

**Comment by coder2020official** [comment]
Example of class-based middlewares:
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/middleware/class_based/basic_example.py

And, instead of using register_next_step_handler try using states:
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/custom_states.py
